File: server/router/cmdb/workflow.go
package cmdb

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/gin-gonic/gin"
)

type WorkFlowRouter struct {
}

func (a *WorkFlowRouter) InitWorkFlowRouter(Router *gin.RouterGroup) {
	workFlowRouter := Router.Group("workflow")
	workflowTemplateApi := v1.ApiGroupApp.CmdbApiGroup.WorkFlowTemplateApi
	{
		workFlowRouter.GET("getTemplateList", workflowTemplateApi.GetWorkFlowTemplateList)  // 查询工作流模板
		workFlowRouter.POST("createTemplate", workflowTemplateApi.CreateWorkFlowTemplate)   // 创建工作流模板
		workFlowRouter.POST("updateTemplate", workflowTemplateApi.UpdateWorkFlowTemplate)   // 更新工作流模板
		workFlowRouter.DELETE("deleteTemplate", workflowTemplateApi.DeleteWorkFlowTemplate) // 删除工作流模板
		workFlowRouter.POST("getTemplateById", workflowTemplateApi.GetWorkFlowTemplateById)

		// 流程状态
		workFlowRouter.GET("getStatusList", workflowTemplateApi.GetTemplateStatus)
		workFlowRouter.POST("createStatus", workflowTemplateApi.CreateTemplateStatus)
		workFlowRouter.POST("updateStatus", workflowTemplateApi.UpdateTemplateStatus)
		workFlowRouter.DELETE("deleteStatus", workflowTemplateApi.DeleteTemplateStatus)
		workFlowRouter.POST("getStatusById", workflowTemplateApi.GetTemplateStatusById)

		// 流程流转
		workFlowRouter.GET("getCircleList", workflowTemplateApi.GetCircleList)
		workFlowRouter.POST("createCircle", workflowTemplateApi.CreateCircle)
		workFlowRouter.POST("updateCircle", workflowTemplateApi.UpdateCircle)
		workFlowRouter.DELETE("deleteCircle", workflowTemplateApi.DeleteCircle)
		workFlowRouter.POST("getCircleById", workflowTemplateApi.GetCircleById)
	}

	workflowOrderApi := v1.ApiGroupApp.CmdbApiGroup.WorkflowOrderApi
	{
		workFlowRouter.POST("createOrder", workflowOrderApi.CreateOrder)
		workFlowRouter.GET("getOrderList", workflowOrderApi.GetOrderList)
		workFlowRouter.POST("updateOrder", workflowOrderApi.UpdateOrder)
		workFlowRouter.DELETE("deleteOrder", workflowOrderApi.DeleteOrder)
		workFlowRouter.POST("getOrderDetailById", workflowOrderApi.GetOrderById)
		workFlowRouter.POST("handleOrder", workflowOrderApi.HandleOrder)
	}

}

File: web/src/api/workflow.js
import service from '@/utils/request'

export const getWorkflowTemplateList = (params) => {
  return service({
    url: '/workflow/getTemplateList',
    method: 'get',
    params: params
  })
}

export const createWorkflowTemplate = (data) => {
  return service({
    url: '/workflow/createTemplate',
    method: 'post',
    data: data,
  })
}

export const deleteWorkflowTemplate = (data) => {
  return service({
    url: '/workflow/deleteTemplate',
    method: 'delete',
    data: data,
  })
}

export const updateWorkflowTemplate = (data) => {
  return service({
    url: '/workflow/updateTemplate',
    method: 'post',
    data: data,
  })
}

export const getWorkflowTemplateById = (data) => {
  return service({
    url: '/workflow/getTemplateById',
    method: 'post',
    data: data,
  })
}

export const getTemplateStatusList = (params) => {
  return service({
    url: '/workflow/getStatusList',
    method: 'get',
    params: params
  })
}

export const createTemplateStatus = (data) => {
  return service({
    url: '/workflow/createStatus',
    method: 'post',
    data: data,
  })
}

export const deleteTemplateStatus = (data) => {
  return service({
    url: '/workflow/deleteStatus',
    method: 'delete',
    data: data,
  })
}

export const updateTemplateStatus = (data) => {
  return service({
    url: '/workflow/updateStatus',
    method: 'post',
    data: data,
  })
}

export const getTemplateStatusById = (data) => {
  return service({
    url: '/workflow/getStatusById',
    method: 'post',
    data: data,
  })
}

export const getCircleList = (params) => {
  return service({
    url: '/workflow/getCircleList',
    method: 'get',
    params: params
  })
}

export const createCircle = (data) => {
  return service({
    url: '/workflow/createCircle',
    method: 'post',
    data: data,
  })
}

export const deleteCircle = (data) => {
  return service({
    url: '/workflow/deleteCircle',
    method: 'delete',
    data: data,
  })
}

export const updateCircle = (data) => {
  return service({
    url: '/workflow/updateCircle',
    method: 'post',
    data: data,
  })
}

export const getCircleById = (data) => {
  return service({
    url: '/workflow/getCircleById',
    method: 'post',
    data: data,
  })
}

export const getOrderList = (params) => {
  return service({
    url: '/workflow/getOrderList',
    method: 'get',
    params: params
  })
}

export const getOrderDetail = (data) => {
  return service({
    url: '/workflow/getOrderDetailById',
    method: 'post',
    data: data
  })
}

export const createWorkflowOrder = (data) => {
  return service({
    url: '/workflow/createOrder',
    method: 'post',
    data: data,
  })
}
